lesson_10/process_sales_gcp.py

```diff
@@ -0,0 +1,88 @@
+import os
+import shutil
+import json
+from datetime import datetime
+import pandas as pd
+import requests
+from airflow import DAG
+from airflow.operators.python import PythonOperator
+from airflow.models import Variable
+from airflow.providers.google.cloud.transfers.local_to_gcs import LocalFilesystemToGCSOperator
+from airflow import macros
+
+
+BASE_DIR = Variable.get('SALES_BASE_DIR')
+AUTH_TOKEN = Variable.get('secret_fake_api_token')
+REQUEST_TIMEOUT_SECONDS = 10
+DEFAULT_ARGS = {
+    'depends_on_past': False,
+    'email_on_failure': False,
+    'email_on_retry': False,
+    'retries': 1,
+    'retry_delay': 30,
+    'max_active_runs': 1,
+}
+
+
+def run_extract_job(raw_dir: str, ds: str) -> None:
+    '''
+        Loads sales from https://fake-api-vycpfa6oca-uc.a.run.app/sales and 
+            saves them to raw_dir in .csv format. All data in raw_dir will be rewrited
+        Params:
+            ds - date in "yyyy-mm-dd" format
+            raw_dir - dir to save
+    '''
+    page = 1
+    records = []
+    while True:
+        # request current page
+        response = requests.get(
+            url='https://fake-api-vycpfa6oca-uc.a.run.app/sales',
```

Comment:
It is better to store URL parameters as function parameters and pass

Reply:
done